action/recipes/recipes.go

```diff
@@ -132,31 +138,78 @@ func Build(
 	// Check results
 	err = nil
 	for _, item := range builds {
-		if item.BuildResult != nil && !errors.Is(item.BuildResult, ErrBuildSkipped) {
+		if item.BuildResult != nil && !errors.Is(item.BuildResult, ErrBuildUpToDate) {
 			err = item.BuildResult
 		}
 	}
 
 	return builds, err
 }
 
+// IsDirEmpty returns whether given directory is empty or not
+func IsDirEmpty(path string) (bool, error) {
+	// Source: https://stackoverflow.com/questions/30697324/how-to-check-if-directory-on-path-is-empty
+	f, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	defer f.Close()
+
+	// File.Readdirnames() take a parameter which is used to limit the number of returned values
+	// It is enough to query only 1 child
+	// File.Readdirnames() is faster than File.Readdir()
+	_, err = f.Readdirnames(1)
+	if err == io.EOF {
+		return true, nil
+	}
+	return false, err // Either not empty or error, suits both cases
+}
+
 // Execute a build step
+// func Execute(ctx context.Context, target string, config *Config, interactive bool, bulldozeMode bool) error {
 func Execute(ctx context.Context, target string, config *Config, interactive bool) error {
-	// Setup dagger client
-	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stdout))
+	// Prep
+	_, err := os.Stat(TimestampsDir)
 	if err != nil {
-		return err
+		err = os.MkdirAll(TimestampsDir, os.ModePerm)
```

Comment:
Same here, you can save yourself the call to `os.Stat` as `os.MkdirAll`  does not return an error on an existing directory